src/Components/NumberOfPets.js

```diff
@@ -1,19 +1,58 @@
-import React from "react";
+import React, { useState } from "react";
 import "./NumberOfPets.css";
 
-class NumberOfPets extends React.Component {
-  render() {
-    return (
-      <section className="num-of-pets">
+export default function NumberOfPets() {
+  const [ count, setCount ] = useState(0);
+
+  const handleIncrement = () => {setCount(count + 1)}
+
+  const handleDecrement = () => {setCount(count <= 0 ? count : count - 1)}
```

Comment:
```suggestion
  const handleDecrement = () => {
    if (count > 0) setCount( count - 1 )
  }
```
Similarly to your `handleSubmit` event handler in AnimalTypes.js, you're potentially causing unnecessary re-renders here. In general, you want to avoid unnecessary re-renders as much as possible because they can slow down your app. 
You're currently using a ternary operator _inside_ of `setCount`, which makes it so that `setCount` gets called whether you need it or not.  By writing it this way instead, using a regular if statement _outside_ of `setCount,` you **only** call `setCount` if it's actually going to do something. If you want the count to remain unchanged, you simply use an if statement to prevent `setCount` from being called at all, which avoids any unnecessary re-renders.